### PR TITLE
Check `$objPage` in `Controller::getTemplate()`

### LINKED ISSUE #3928: $objPage could be null
### Affected version(s)

4.x

### Description

Hello,

using Contao as API-Framework the $objPage could be null at 
https://github.com/contao/contao/blob/4166f4eeccafb9cdcbd15e5e89728bbb4f1f3a81/core-bundle/src/Resources/contao/library/Contao/Controller.php#L82

in my case i´m using the notificationCenter to send emails during an API-Request and there is a warning because $objPage is null. 

So maybe we can check if $objPage !== null

Thanks and Greetings

### PR DESCRIPTION
**Pull request by xprojects-de**
Fixes https://github.com/contao/contao/issues/3928

### CODE REVIEW

File: core-bundle/src/Resources/contao/library/Contao/Controller.php [hunk 1]
```diff
@@ -77,7 +77,7 @@ public static function getTemplate($strTemplate)
 			/** @var PageModel $objPage */
 			global $objPage;
 
-			if ($objPage->templateGroup)
+			if (null !== $objPage && $objPage->templateGroup)
```
Comment (m-vo) [inline core-bundle/src/Resources/contao/library/Contao/Controller.php]: The typehint three lines above should also be adjusted to `/** @var PageModel|null $objPage */`.

**Review by m-vo — CHANGES_REQUESTED** [review]

File: core-bundle/src/Resources/contao/library/Contao/Controller.php (hunk 1 above)
Comment (fritzmg) [reply core-bundle/src/Resources/contao/library/Contao/Controller.php]: That would need to be done everywhere though (there are 40 more occurrences). Imho that should be a separate PR.

**Review by m-vo — DISMISSED** [review]

File: core-bundle/src/Resources/contao/library/Contao/Controller.php (hunk 1 above)
Comment (xprojects-de) [reply core-bundle/src/Resources/contao/library/Contao/Controller.php]: Done. It also makes sense there...

**Review by bytehead — DISMISSED** [review]

File: core-bundle/src/Resources/contao/library/Contao/Controller.php [hunk 2]
```diff
@@ -74,10 +74,10 @@ public static function getTemplate($strTemplate)
 		// Check for a theme folder
 		if (\defined('TL_MODE') && TL_MODE == 'FE')
 		{
-			/** @var PageModel $objPage */
+			/** @var PageModel|null $objPage */
 			global $objPage;
 
-			if ($objPage->templateGroup)
+			if (null !== $objPage && $objPage->templateGroup)
```
Comment (leofeyer) [inline core-bundle/src/Resources/contao/library/Contao/Controller.php]:
```suggestion
			if ($objPage->templateGroup ?? null)
```

Comment (xprojects-de) [reply core-bundle/src/Resources/contao/library/Contao/Controller.php]: Does in this case the null coalescing Operator not crash if objPage itself is null?

Comment (fritzmg) [reply core-bundle/src/Resources/contao/library/Contao/Controller.php]: No it won't crash: https://3v4l.org/hlVR1 :)

Comment (xprojects-de) [reply core-bundle/src/Resources/contao/library/Contao/Controller.php]: Crazy but does what it's supposed to ... Never used the Coalescing Operator in this way :-) "Again what learned" (as Lothar would say ). Thanks

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @xprojects-de.

**Pull request merged**